proxy/src/auth.rs

```diff
@@ -6,7 +6,7 @@ use crate::config::{AuthBackendType, ProxyConfig};
 use crate::error::UserFacingError;
 use crate::stream::PqStream;
 use crate::{auth_backend, compute, waiters};
-use console::ConsoleAuthError::SniMissing;
+use console::ConsoleAuthError::SniMissingAndProjectNameMissing;
```

Comment:
I'd suggest just "ProjectNameMissing". We might have other ways of figuring out the project name in the future, and we don't want this to become "SniAndProjectNameAndBlahBlahMissing". At the end of the day, the error is that we couldn't figure out the project name, not from the SNI or the new option.

Reply:
This is already known as [`ProjectNameError`](https://github.com/neondatabase/neon/blob/36281e3b475ac46570dd4f89a61fc525ff3f0a1c/proxy/src/auth/credentials.rs#L39). The patch should be rebased.